operator/pkg/helmreconciler/prune.go

```diff
@@ -139,9 +139,11 @@ func (h *HelmReconciler) DeleteObjectsList(objectsList []*unstructured.Unstructu
 				continue
 			}
 
-			err := h.client.Delete(context.TODO(), &o, client.PropagationPolicy(metav1.DeletePropagationBackground))
+			err := h.client.Delete(context.TODO(), &o,
+				client.PropagationPolicy(metav1.DeletePropagationBackground), client.GracePeriodSeconds(0))
 			if err != nil {
-				if !strings.Contains(err.Error(), "not found") {
+				if !strings.Contains(err.Error(), "not found") &&
+					!strings.Contains(err.Error(), "could not find the requested resource") {
```

Comment:
maybe we should dedupe the list so we don't try to remove twice instead of hiding the error? if I am understanding this right

Reply:
it is not about duplicate, but instead related to user experience, if users already removed some of the resources manually, we don't want to fail the command, but instead issuing a warning message should be enough.

Reply:
Are you sure? when I run this it is clearly duplicating resources. It seems extremely unlikely that a user is running `kubectl delete` at the exact same time they run `istioctl uninstall`. 

I think the change is good but we are masking a real bug?
```
$ ik x uninstall --purge
All Istio resources will be pruned from the cluster
Proceed? (y/N) y
  Removed HorizontalPodAutoscaler:istio-system:istio-ingressgateway.
  Removed HorizontalPodAutoscaler:istio-system:istiod.
  Removed PodDisruptionBudget:istio-system:istio-ingressgateway.
  Removed PodDisruptionBudget:istio-system:istiod.
  Removed Deployment:istio-system:istio-ingressgateway.
  Removed Deployment:istio-system:istiod.
  Removed Deployment:istio-system:prometheus.
  Removed Service:istio-system:istio-ingressgateway.
  Removed Service:istio-system:istiod.
  Removed Service:istio-system:prometheus.
  Removed ConfigMap:istio-system:istio.
  Removed ConfigMap:istio-system:istio-sidecar-injector.
  Removed ConfigMap:istio-system:prometheus.
  Removed ServiceAccount:istio-system:istio-ingressgateway-service-account.
  Removed ServiceAccount:istio-system:istio-reader-service-account.
  Removed ServiceAccount:istio-system:istiod-service-account.
  Removed ServiceAccount:istio-system:prometheus.
  Removed RoleBinding:istio-system:istio-ingressgateway-sds.
  Removed RoleBinding:istio-system:istiod-istio-system.
  Removed Role:istio-system:istio-ingressgateway-sds.
  Removed Role:istio-system:istiod-istio-system.
  Removed EnvoyFilter:istio-system:metadata-exchange-1.6.
  Removed EnvoyFilter:istio-system:metadata-exchange-1.7.
  Removed EnvoyFilter:istio-system:stats-filter-1.6.
  Removed EnvoyFilter:istio-system:stats-filter-1.7.
  Removed EnvoyFilter:istio-system:tcp-metadata-exchange-1.6.
  Removed EnvoyFilter:istio-system:tcp-metadata-exchange-1.7.
  Removed EnvoyFilter:istio-system:tcp-stats-filter-1.6.
  Removed EnvoyFilter:istio-system:tcp-stats-filter-1.7.
  Removed MutatingWebhookConfiguration::istio-sidecar-injector.
  Removed ValidatingWebhookConfiguration::istiod-istio-system.
  Removed ClusterRole::istio-reader-istio-system.
  Removed ClusterRole::istiod-istio-system.
  Removed ClusterRole::prometheus-istio-system.
  Removed ClusterRoleBinding::istio-reader-istio-system.
  Removed ClusterRoleBinding::istiod-istio-system.
  Removed ClusterRoleBinding::istiod-pilot-istio-system.
  Removed ClusterRoleBinding::prometheus-istio-system.
object: MutatingWebhookConfiguration::istio-sidecar-injector is not being deleted because it no longer exists
  Removed MutatingWebhookConfiguration::istio-sidecar-injector.
object: ValidatingWebhookConfiguration::istiod-istio-system is not being deleted because it no longer exists
  Removed ValidatingWebhookConfiguration::istiod-istio-system.
  Removed CustomResourceDefinition::adapters.config.istio.io.
  Removed CustomResourceDefinition::attributemanifests.config.istio.io.
  Removed CustomResourceDefinition::authorizationpolicies.security.istio.io.
  Removed CustomResourceDefinition::destinationrules.networking.istio.io.
  Removed CustomResourceDefinition::envoyfilters.networking.istio.io.
  Removed CustomResourceDefinition::gateways.networking.istio.io.
  Removed CustomResourceDefinition::handlers.config.istio.io.
  Removed CustomResourceDefinition::httpapispecbindings.config.istio.io.
  Removed CustomResourceDefinition::httpapispecs.config.istio.io.
  Removed CustomResourceDefinition::instances.config.istio.io.
  Removed CustomResourceDefinition::istiooperators.install.istio.io.
  Removed CustomResourceDefinition::peerauthentications.security.istio.io.
  Removed CustomResourceDefinition::quotaspecbindings.config.istio.io.
  Removed CustomResourceDefinition::quotaspecs.config.istio.io.
  Removed CustomResourceDefinition::requestauthentications.security.istio.io.
  Removed CustomResourceDefinition::rules.config.istio.io.
  Removed CustomResourceDefinition::serviceentries.networking.istio.io.
  Removed CustomResourceDefinition::sidecars.networking.istio.io.
  Removed CustomResourceDefinition::templates.config.istio.io.
  Removed CustomResourceDefinition::virtualservices.networking.istio.io.
  Removed CustomResourceDefinition::workloadentries.networking.istio.io.
  Removed CustomResourceDefinition::workloadgroups.networking.istio.io.
object: CustomResourceDefinition::adapters.config.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::adapters.config.istio.io.
object: CustomResourceDefinition::attributemanifests.config.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::attributemanifests.config.istio.io.
object: CustomResourceDefinition::authorizationpolicies.security.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::authorizationpolicies.security.istio.io.
object: CustomResourceDefinition::destinationrules.networking.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::destinationrules.networking.istio.io.
object: CustomResourceDefinition::envoyfilters.networking.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::envoyfilters.networking.istio.io.
object: CustomResourceDefinition::gateways.networking.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::gateways.networking.istio.io.
object: CustomResourceDefinition::handlers.config.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::handlers.config.istio.io.
object: CustomResourceDefinition::httpapispecbindings.config.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::httpapispecbindings.config.istio.io.
object: CustomResourceDefinition::httpapispecs.config.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::httpapispecs.config.istio.io.
object: CustomResourceDefinition::instances.config.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::instances.config.istio.io.
object: CustomResourceDefinition::istiooperators.install.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::istiooperators.install.istio.io.
object: CustomResourceDefinition::peerauthentications.security.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::peerauthentications.security.istio.io.
object: CustomResourceDefinition::quotaspecbindings.config.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::quotaspecbindings.config.istio.io.
object: CustomResourceDefinition::quotaspecs.config.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::quotaspecs.config.istio.io.
object: CustomResourceDefinition::requestauthentications.security.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::requestauthentications.security.istio.io.
object: CustomResourceDefinition::rules.config.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::rules.config.istio.io.
object: CustomResourceDefinition::serviceentries.networking.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::serviceentries.networking.istio.io.
object: CustomResourceDefinition::sidecars.networking.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::sidecars.networking.istio.io.
object: CustomResourceDefinition::templates.config.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::templates.config.istio.io.
object: CustomResourceDefinition::virtualservices.networking.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::virtualservices.networking.istio.io.
object: CustomResourceDefinition::workloadentries.networking.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::workloadentries.networking.istio.io.
object: CustomResourceDefinition::workloadgroups.networking.istio.io is not being deleted because it no longer exists
  Removed CustomResourceDefinition::workloadgroups.networking.istio.io.
✔ Uninstall complete
```

Reply:
let me double check this before merging

Reply:
yep I think the reason is that for CRD we have two versions: v1/v1beta1, but during Listing seems like the version does not matter, so it would return two sets of CRDs